packages/block-library/src/navigation-link/edit.js

```diff
@@ -133,6 +133,53 @@ function getSuggestionsQuery( type, kind ) {
 	}
 }
 
+export const updateNavigationLinkBlockAttributes = (
+	updatedValue = {},
+	{
+		setAttributes,
+		label: originalLabel = '',
+		kind: originalKind = '',
+		type: originalType = '',
+	}
+) => {
+	const {
+		title = '',
+		url = '',
+		opensInNewTab,
+		id,
+		kind: newKind = originalKind,
+		type: newType = originalType,
+	} = updatedValue;
+
+	const normalizedTitle = title.replace( /http(s?):\/\//gi, '' );
+	const normalizedURL = url.replace( /http(s?):\/\//gi, '' );
+	const escapeTitle =
+		title !== '' &&
+		normalizedTitle !== normalizedURL &&
+		originalLabel !== title;
+	const label = escapeTitle
+		? escape( title )
+		: originalLabel || escape( normalizedURL );
+
+	const type = newType === 'post_tag' ? 'tag' : newType.replace( '-', '_' );
+
+	const isBuiltInType =
+		[ 'post', 'page', 'tag', 'category' ].indexOf( type ) > -1;
+
+	const isCustomLink =
+		( ! newKind && ! isBuiltInType ) || newKind === 'custom';
+	const kind = isCustomLink ? 'custom' : newKind;
+
+	return setAttributes( {
+		url: encodeURI( url ),
+		label,
+		opensInNewTab,
+		...( Number.isInteger( id ) && { id } ),
```

Comment:
I double checked what gets returned for a `post-format` type, from `/wp/v2/search` and it's a string value vs int, so this won't get saved in that case.

<img width="669" alt="Screen Shot 2021-05-03 at 12 56 19 PM" src="https://user-images.githubusercontent.com/1270189/116926286-0d25cc80-ac0f-11eb-8a11-1c9c7d8f11bf.png">

Reply:
Yep. I have a dedicated issue open for this.

Reply:
> Yep. I have a dedicated issue open for this.

Excellent, thanks @getdave

Reply:
Here is the Issue https://github.com/WordPress/gutenberg/issues/31380